### DISCUSSION TITLE
.sqlmodel_update() raise RuntimeError: dictionary changed size during iteration

### DISCUSSION THREAD

**Discussion by adnankaya** [fastapi/sqlmodel#982] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
def foo(hero_in:HeroCreate, related_id:int, owner_id:int):
    hero.sqlmodel_update(
            hero_in,
            update={
                "related_id": related_id,
                "owner_id": owner_id,
            },
        )

#####
def sqlmodel_update(
        self: _TSQLModel,
        obj: Union[Dict[str, Any], BaseModel],
        *,
        update: Union[Dict[str, Any], None] = None,
    ) -> _TSQLModel:
        use_update = (update or {}).copy()
        if isinstance(obj, dict):
            for key, value in {**obj, **use_update}.items():
                if key in get_model_fields(self):
                    setattr(self, key, value)
        elif isinstance(obj, BaseModel):
            for key in get_model_fields(obj):
                if key in use_update:
                    value = use_update.pop(key)
                else:
                    value = getattr(obj, key)
                setattr(self, key, value)
            for remaining_key in use_update:
                if remaining_key in get_model_fields(self):
                    value = use_update.pop(remaining_key) # NOTE the problem is here.
                    setattr(self, remaining_key, value)
        else:
            raise ValueError(
                "Can't use sqlmodel_update() with something that "
                f"is not a dict or SQLModel or Pydantic model: {obj}"
            )
        return self
```

### Description

Hi @tiangolo , thanks for sqlmodel. 
Related to PR https://github.com/tiangolo/sqlmodel/pull/804
When I tried to update my existing db object which is **hero** and its `related_id`, `owner_id` fields I encountered **[RuntimeError: dictionary changed size during iteration](https://stackoverflow.com/questions/11941817/how-can-i-avoid-runtimeerror-dictionary-changed-size-during-iteration-error)**. In the sqlmodel_update function The following part is manipulates the dictionary use_update. That's needed to be fixed I think.
```python
for remaining_key in use_update:
                if remaining_key in get_model_fields(self):
                    value = use_update.pop(remaining_key) # NOTE the problem is here.
                    setattr(self, remaining_key, value)
```
Instead of` value = use_update.pop(remaining_key)` we can use update `value = update.pop(remaining_key)`

### Operating System

macOS

### Operating System Details

Mac Os M2

### SQLModel Version

0.0.16

### Python Version

3.10.14

**Comment by BartSchuurmans** [comment, marked as answer]
I've submitted a fix in https://github.com/tiangolo/sqlmodel/pull/997

**Reply by tiangolo** [reply]
Thank you! This is available in SQLModel 0.0.28, just released. :tada:
